### DISCUSSION TITLE
TMS Maps

### DISCUSSION THREAD

**Discussion by abdullahO2** [organicmaps/organicmaps#2574] [Ideas]
Is it possible to add custom online raster maps (TMS Maps) as an additional layer to the map and add it via URLs like:
https://**/{0}/{1}/{2}.png

Thank you and best regards

**Comment by biodranik** [comment]
No. Why do you need it? What is your use-case?

**Reply by abdullahO2** [reply]
To add upper or lower layers to the map, such as:
RainViewer (Weather Radar App) (Upper Layer,overlay)
https://www.rainviewer.com/api/weather-maps-api.html
https://tilecache.rainviewer.com/v2/radar/latest/512/{z}/{x}/{y}/2/1_0.png

OSM TMS (Underlay Layer,base):
https://tile.openstreetmap.org/${z}/${x}/${y}.png
https://wiki.openstreetmap.org/wiki/Tile_servers

And layers of satellite images such as Bing, Mapbox and others, see:
https://anygis.ru/Web/Html/Download_en?app=Desktop

**Comment by hb0nd** [comment]
The styles are mapcss, so with a bit of work you could host your own tileserver. There's no point in organicmaps as a project generating tiles though

**Reply by abdullahO2** [reply]
What I suggest is that the developer adds a feature that allows the user to add layers as they wish through tiles URL

**Reply by hb0nd** [reply]
> What I suggest is that the developer adds a feature that allows the user to add layers as they wish through tiles URL

Ooh I see. That's up to the devs, but I think it'll probably be out of scope I'm afraid.

**Comment by RicoElectrico** [comment]
I think there's an interesting use case: for those who complain about lack of aerial imagery, they would now have options to hook up any TMS they want, e.g. with a Mapbox token from their personal account, for which the free tier should be enough.
